message/router/middleware/duplicator.go

```diff
@@ -0,0 +1,15 @@
+package middleware
+
+import (
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// Duplicator sends message twice.
+func Duplicator(h message.HandlerFunc) message.HandlerFunc {
+	return func(msg *message.Message) ([]*message.Message, error) {
+		events, err := h(msg)
+		_, _ = h(msg)
```

Comment:
Handling of these errors is an interesting case. If we'd like to make sure that our system is idempotent, we should probably check if the second handler didn't return an error.

Also worth considering if other middlewares won't cause some weird issues.